src/Common/SymbolIndex.h

```diff
@@ -24,7 +24,6 @@ class SymbolIndex : private boost::noncopyable
 
 public:
     static MultiVersion<SymbolIndex>::Version instance();
```

Comment:
Then there is also no sense in using `MultiVersion` here as well, since it had been introduced in #17637 just to implement reload

Cc: @amosbird

Reply:
Yes, let's remove MultiVersion here.
Now we have a strict no-dlopen policy, and we will soon remove shared linking and glibc entirely!

Reply:
#51993